src/Features/CSharp/IntroduceVariable/CSharpIntroduceVariableService_IntroduceLocal.cs

```diff
@@ -59,6 +59,11 @@ internal partial class CSharpIntroduceVariableService
                 return Task.FromResult(IntroduceLocalDeclarationIntoLambda(
                     document, expression, newLocalName, declarationStatement, parentLambda, allOccurrences, cancellationToken));
             }
+            else if (IsInExpressionBodiedMember(expression))
```

Comment:
Yep, I'm generally wary of unqualified `GetAncestorOrThis()` calls because of this.